### ISSUE TITLE
Exception: videoQuality cannot be empty

### ISSUE THREAD

**Issue by mfurkanyuceal** [newtaDev/pod_player#82]
I started getting this error on videos we just uploaded to Vimeo.

Usage:
![image](https://user-images.githubusercontent.com/22933926/198876267-4e81a73b-132f-474b-b908-d027b35c15b7.png)

Sentry issue:
![image](https://user-images.githubusercontent.com/22933926/198876444-8da2d0ce-0330-4ce3-8d4b-6043c5ddbc2e.png)

- Flutter Version =>  3.3.5
- Dart Version => 2.18.2
- Pod Player Version => 0.1.0

**Comment by ikram3mkaissi** [comment]
i'm also getting this error

**Comment by newtaDev** [comment]
@mfurkanyuceal @ikram3mkaissi 
Thank you for filling issue, 
Can u please share your vimeo ids that cause issues.. 
Example app works fine

**Comment by mfurkanyuceal** [comment]
> 
[766152970](https://vimeo.com/766152970)

**Comment by Javierd** [comment]
I'm currently getting this error on new uploaded videos

**Comment by ikhattaab** [comment]
Also having the same issue (videoQuality cannot be empty) with youtube live stream.

**Comment by ikram3mkaissi** [comment]
any solution?

**Comment by dinurymomshad** [comment]
I have the same issue. When can we expect a solution?

**Comment by AmirSarraf** [comment]
For this to work, i think we need a public access token and then we should use `https://api.vimeo.com/videos/{video_id}` instead of `https://player.vimeo.com/video/$videoId/config` to get the video info, am i right?

**Comment by Javierd** [comment]
Are there any updates on this problem?

**Comment by Mobigic-Vaibhav** [comment]
Get the same issue for the link https://www.youtube.com/watch?v=bnw9qGkY810 any update on this.

**Comment by vaibhavmishra571999** [comment]
(Unhandled Exception: Exception: videoQuality cannot be empty)
Having same issue but in youtube videos.

**Comment by chirag-dt** [comment]
Also having the same issue (videoQuality cannot be empty) with vimeo player

**Comment by AyeshaIftikhar** [comment]
> A quick workaround can be used setting an arbitrary value for the quality i chose 720 in this case and pointing to the hls stream can be used for now in the file video_apis.dart replace the code in the try block for getVimeoVideoQualityUrls with the following it will point the first cdn list for the hls streams in the event progressive urls come back empty.
> 
> ```
> 
> 
>   final response = await http.get(
>         Uri.parse('https://player.vimeo.com/video/$videoId/config'),
>       );
>       var jsonData =
>           jsonDecode(response.
>           body)['request']['files']['progressive'];
> 
>       var list=  List.generate(
>         jsonData.length,
>         (index) => VideoQalityUrls(
>           quality: int.parse(
>             (jsonData[index]['quality'] as String?)?.split('p').first ?? '0',
>           ),
>           url: jsonData[index]['url'],
>         ),
>       );
>        if(list.isEmpty){
>          jsonData =  jsonDecode(response.body)['request']['files']['hls']['cdns'];
> 
>          var url = jsonData.entries.first.value["url"];
>          list = [];
> 
>          jsonData = jsonData.entries.map((entry) => "${entry.key} + ${entry.value}").toList();
>         list.add( VideoQalityUrls(
>            quality: 720,
>            url: url,
>          ));
> 
>        }
>        return list;
> ```

Your solution is a life saver.

**Comment by ahmed-moha** [comment]
> 

if you choose that way the video and audio may not play at one time

**Comment by newtaDev** [comment]
Now this issue got fixed `pod_player > 0.1.2`

**Comment by xurc** [comment]
This may have been fixed for Vimeo, but for YouTube streams the bug is still there. For example, with pod_player 0.1.2 for this stream:

https://youtu.be/86YLFOog4GM

I got:
```
E/flutter (26777): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Exception: videoQuality cannot be empty
E/flutter (26777): #0      PodPlayerController._checkAndWaitTillInitialized (package:pod_player/src/controllers/pod_player_controller.dart:73:9)
E/flutter (26777): #1      PodPlayerController._checkAndWaitTillInitialized (package:pod_player/src/controllers/pod_player_controller.dart:82:11)
E/flutter (26777): <asynchronous suspension>
E/flutter (26777): #2      PodPlayerController._checkAndWaitTillInitialized (package:pod_player/src/controllers/pod_player_controller.dart:82:5)
E/flutter (26777): <asynchronous suspension>
E/flutter (26777): #3      PodPlayerController.initialise (package:pod_player/src/controllers/pod_player_controller.dart:61:5)
E/flutter (26777): <asynchronous suspension>
E/flutter (26777): 
```
